apps/scrtdd/hdd/waveform.cpp

```diff
@@ -480,28 +480,38 @@ bool xcorr(const GenericRecordCPtr &tr1,
   };
 
   // do as much computation as possible outside the main xcorr loop
+  double localMeanL = 0;
+  for (int idxS = 0; idxS < smpsSsize; idxS++)
+  {
+    double sample = sampleAtLong(idxS, -(maxDelaySmps + 1));
+    localMeanL += sample;
+  }
+  localMeanL /= smpsSsize;
+
   double denomL = 0, denomS = 0;
   for (int idxS = 0; idxS < smpsSsize; idxS++)
   {
     denomS += smpsS[idxS] * smpsS[idxS];
     double sampleL = sampleAtLong(idxS, -(maxDelaySmps + 1));
-    denomL += sampleL * sampleL;
+    denomL += (sampleL - localMeanL) * (sampleL - localMeanL);
   }
 
   // cross-correlation loop
   for (int delay = -maxDelaySmps; delay < maxDelaySmps; delay++)
   {
     // remove from denomL the sample that has exited the current xcorr win
     double lastSampleL = sampleAtLong(-1, delay);
-    denomL -= lastSampleL * lastSampleL;
+    localMeanL         = (localMeanL * smpsSsize - lastSampleL) / smpsSsize;
+    denomL -= (lastSampleL - localMeanL) * (lastSampleL - localMeanL);
```

Comment:
Unfortunately this is not correct. At a new iteration the values used to compute the previous denomL are not valid anymore (the mean has changed), so we should recompute the denomL in full

Reply:
Just to be clear: at each iteration denomL is the sum of the squared of each sample-mean
Since at each new iteration the mean changes, that invalidate *all* the previous values